### PR TITLE
ENH: colorize errors v2

### PR DESCRIPTION
**Pull request by RDoerfel**
Create class to colorize keayword in terminal messages by:
- qDebug()
- qWarning()
- qInfo()
- qCritical()
- qFatal()

### CODE REVIEW

File: libraries/utils/generics/applicationlogger.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            LOG_WRITE(std::cout, COLOR_CRITICAL, "CRIT", msg.toStdString());
+            break;
+        case QtFatalMsg:
+            LOG_WRITE(std::cout, COLOR_FATAL, "FATAL", msg.toStdString());
+            break;
+        case QtDebugMsg:
+            LOG_WRITE(std::cout, COLOR_DEBUG, "DEBUG", msg.toStdString());
+            break;
+        case QtInfoMsg:
+            LOG_WRITE(std::cout, COLOR_INFO, "Info", msg.toStdString());
```
Comment (LorenzE) [inline libraries/utils/generics/applicationlogger.cpp]: Info in capital > INFO

File: examples/ex_read_raw/main.cpp [hunk 2]
```diff
@@ -179,7 +182,13 @@ int main(int argc, char *argv[])
 
     if (current_comp != dest_comp)
     {
-        qDebug() << "This part needs to be debugged";
+        qDebug() << "Debug Message";
```
Comment (LorenzE) [inline examples/ex_read_raw/main.cpp]: Remove from example. As a side note, `qFatal` will stop the application immediately.

File: libraries/utils/generics/applicationlogger.cpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+//*************************************************************************************************************
+//=============================================================================================================
+// Definitions
+//=============================================================================================================
+
+#ifdef WIN32
+    #include <wchar.h>
+    #include <windows.h>
+    #define COLOR_INFO          SetConsoleTextAttribute(hOut, 0x0A)     //green
+    #define COLOR_DEBUG         SetConsoleTextAttribute(hOut, 0x02)     //green
```
[the remark this answers is not in the document]
Comment (RDoerfel) [reply libraries/utils/generics/applicationlogger.cpp]: This are different greens. I could change it to the same, if you want.

File: libraries/utils/generics/applicationlogger.h [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+/**
+ * Color command line outputs
+ *
+ * @brief Apllicationlogger
+ */
+class UTILSSHARED_EXPORT ApplicationLogger
+{
+public:
+
+    ApplicationLogger();
```
[the remark this answers is not in the document]
Comment (RDoerfel) [reply libraries/utils/generics/applicationlogger.h]: error: 'ApplicationLogger' is not virtual and cannot be declared pure

File: libraries/utils/generics/applicationlogger.cpp [hunk 5]
```diff
@@ -0,0 +1,129 @@
+//=============================================================================================================
+/**
+ * @file     applicationlogger.h
+ * @author   Mainak Jas <mjas@mgh.harvard.edu>;
+ *           Ruben Dörfel <rdorfel@mgh.harvard.edu>
```
Comment (LorenzE) [inline libraries/utils/generics/applicationlogger.cpp]: is this our correct email address?

File: libraries/utils/generics/applicationlogger.cpp [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+    #define LOG_WRITE(OUTPUT, COLOR, LEVEL, MSG) OUTPUT << COLOR << "" LEVEL " " << COLOR_RESET << MSG << "\n"
+#endif
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+void ApplicationLogger::CustomLogWriter(QtMsgType type,
```
Comment (LorenzE) [inline libraries/utils/generics/applicationlogger.cpp]: functions should start with a lower case: `customLogWriter`

File: libraries/utils/generics/applicationlogger.h [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+// DEFINE NAMESPACE UTILSLIB
+//=============================================================================================================
+
+namespace UTILSLIB
+{
+//=============================================================================================================
+/**
+ * The Apllicationlogger provides colorized keywords in terminal outputs for better overview
+ *
+ * @brief Apllicationlogger
```
Comment (LorenzE) [inline libraries/utils/generics/applicationlogger.h]: typo should be `ApplicationLogger`

File: libraries/utils/generics/applicationlogger.h [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    //=========================================================================================================
+    /**
+     * Customized logWriter to colorize type of the message in the terminal
+     *
+     * @param[in]  type      The type of the qMessage, can be qDebug, qInfo, qCritical, qWarning and qFatal
+     * @param[in]  context   Context provides information about the source code location
+     * @param[out]  msg      The message to print in the terminal
+     *
+     */
+
```
Comment (LorenzE) [inline libraries/utils/generics/applicationlogger.h]: remove empty line

**Pull request merged**